api/middleware/apd.js

```diff
@@ -0,0 +1,108 @@
+const logger = require('../logger')('apd middleware');
+const {
+  apd: ApdModel,
+  apdActivity: defaultActivityModel
+} = require('../db').models;
+
+const cache = {};
+
+module.exports.loadApd = (model = ApdModel, idParam = 'id') => {
+  if (!cache[['loadApd', model, idParam]]) {
+    const loadApd = async (req, res, next) => {
+      logger.silly(req, 'loading APD for request');
+      try {
+        if (model === ApdModel) {
+          req.meta.apd = await ApdModel.where({
+            id: +req.params[idParam]
+          }).fetch({
+            withRelated: ['activities']
+          });
+          next();
+        } else {
+          const obj = await model
+            .where({ id: req.params[idParam] })
+            .fetch({ withRelated: ['apd.activities'] });
+          if (obj) {
+            req.meta.apd = obj.related('apd');
+          }
+          next();
+        }
+      } catch (e) {
+        logger.error(req, e);
+        next();
+      }
+    };
+    cache[['loadApd', model, idParam]] = loadApd;
```

Comment:
consider pulling this (`['loadApd', model, idParam]`) out into a variable (`lookup` or `key`?) since we use a few times

Reply:
👍